### PR TITLE
feat: allow empty auth_username if auth_token is a bearer token

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

**Comment by michielbdejong** [comment]
Included this proposal in https://github.com/interledger/interledger/wiki/Interledger-over-BTP#changes-between-versions-17q4-and-18q1

File: 0023-bilateral-transfer-protocol/0023-bilateral-transfer-protocol.md [hunk 1]
```diff
@@ -186,11 +186,13 @@ Before anything else, when a client connects to a server, it sends a special
 `Message` request. Its primary `protocolData` entry MUST have name `'auth'`,
 content type `MIME_APPLICATION_OCTET_STREAM`,
 and empty data, and among the secondary entries, there MUST be a UTF-8
-`'auth_token'` entry, and a UTF-8 `'auth_username'` entry. The further secondary
+`'auth_token'` entry, and there MAY be a UTF-8 `'auth_username'` entry. The further secondary
```
Comment (dappelt) [inline 0023-bilateral-transfer-protocol/0023-bilateral-transfer-protocol.md]: These two statements contradict each other: 
> [...] there MAY be a UTF-8 `'auth_username'` entry.

and later 

>[...] `'auth_username'` data can be set to the empty string, but they cannot be omitted.

Comment (michielbdejong) [reply 0023-bilateral-transfer-protocol/0023-bilateral-transfer-protocol.md]: ah, good catch! fixed.

**Comment by sharafian** [comment]
Even in `ilp-plugin-btp` where we ordinarily don't specify an `auth_username`, the `auth_username` protocol is still sent to the server. To reflect the current behavior you could add a note saying that the `auth_username` may be an empty string

**Comment by michielbdejong** [comment]
Will merge once https://circleci.com/gh/interledger/rfcs/tree/master is fixed

**Pull request merged**